### PR TITLE
core: use qualname in beta message

### PR DESCRIPTION
**Pull request by efriis**

### CODE REVIEW

File: libs/core/langchain_core/_api/beta_decorator.py [hunk 1]
```diff
@@ -189,10 +190,10 @@ def finalize(wrapper: Callable[..., Any], new_doc: str) -> Any:
                 )
 
         else:
+            _name = _name or obj.__qualname__
             if not _obj_type:
-                _obj_type = "function"
+                _obj_type = "function" if "." not in _name else "method"
```
Comment (efriis) [inline libs/core/langchain_core/_api/beta_decorator.py:195]: edge case this won't work on: a function defined within another function within a test

**Comment by baskaryan** [comment]
lgtm, would add to deprecated

**Comment by efriis** [comment]
Was added to deprecated in #20578 also
